packages/trpc/server/routers/viewer/slots/handleNotificationWhenNoSlots.ts

```diff
@@ -0,0 +1,128 @@
+import type { Dayjs } from "@calcom/dayjs";
+import { sendOrganizationAdminNoSlotsNotification } from "@calcom/emails";
+import { RedisService } from "@calcom/features/redis/RedisService";
+import { IS_PRODUCTION, WEBAPP_URL } from "@calcom/lib/constants";
+import { getTranslation } from "@calcom/lib/server/i18n";
+import { prisma } from "@calcom/prisma";
+
+type EventDetails = {
+  username: string;
+  eventSlug: string;
+  startTime: Dayjs;
+  visitorTimezone?: string;
+  visitorUid?: string;
+};
+
+// Incase any updates are made - lets version the key so we can invalidate
+const REDIS_KEY_VERSION = "V1";
+
+// 7 days or 60s in dev
+const NO_SLOTS_NOTIFICATION_FREQUENCY = IS_PRODUCTION ? 7*24*3600 : 60;
+
+const NO_SLOTS_COUNT_FOR_NOTIFICATION = 2;
+
+const constructRedisKey = (eventDetails: EventDetails, orgSlug?: string) => {
+  return `${REDIS_KEY_VERSION}.${eventDetails.username}:${eventDetails.eventSlug}${
+    orgSlug ? `@${orgSlug}` : ""
+  }`;
+};
+
+const constructDataHash = (eventDetails: EventDetails) => {
+  const obj = {
+    startTime: eventDetails.startTime.format("YYYY-MM-DD"),
+    visitorTimezone: eventDetails?.visitorTimezone,
+    visitorUid: eventDetails?.visitorUid,
+  };
+
+  return JSON.stringify(obj);
+};
+
+export const handleNotificationWhenNoSlots = async ({
+  eventDetails,
+  orgDetails,
+}: {
+  eventDetails: EventDetails;
+  orgDetails: { currentOrgDomain: string | null; isValidOrgDomain: boolean };
```

Comment:
```suggestion
  orgDetails: { currentOrgDomain: string | null;};
```

`isValidOrgDomain` doesn't seem to be used